### PR TITLE
UCAI OpenAI toolkit support

### PR DESCRIPTION
**Pull request by serena-ruan**
**PR Checklist**

- [ ] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**
Support UC functions as OpenAI tools

### CODE REVIEW

File: .github/workflows/ucai-integration-tests.yml
Comment (BenWilson2) [inline .github/workflows/ucai-integration-tests.yml:62]:
```suggestion
  anthropic_test:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ['3.9', '3.10']
    timeout-minutes: 20
    defaults:
      run:
        working-directory: unitycatalog-ai
    steps:
      - name: Checkout code
        uses: actions/checkout@v4
      - name: Set up Python
        uses: actions/setup-python@v5
        with:
          python-version: ${{ matrix.python-version }}
      - name: Install dependencies
        run: |
          pip install .
          pip install integrations/anthropic[dev]
      - name: Run tests
        run: |
          pytest integrations/anthropic/tests

  openai_test:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ['3.9', '3.10']
    timeout-minutes: 20
    defaults:
      run:
        working-directory: unitycatalog-ai
    steps:
      - name: Checkout code
        uses: actions/checkout@v4
      - name: Set up Python
        uses: actions/setup-python@v5
        with:
          python-version: ${{ matrix.python-version }}
      - name: Install dependencies
        run: |
          pip install .
          pip install integrations/openai[dev]
      - name: Run tests
        run: |
          pytest integrations/openai/tests
```

**Comment by BenWilson2** [comment]
@dennyglee The suggestion above should fix the merge conflict (hopefully). I don't have write access to Serena's fork and she's OOO for a bit. Could this get resolved by a maintainer? Thanks!

**Comment by dennyglee** [comment]
@BenWilson2 could you check if this workflow is missing a workflow scope?

File: .github/workflows/ucai-integration-tests.yml
Comment (BenWilson2) [inline .github/workflows/ucai-integration-tests.yml:62]:
```suggestion

  anthropic_test:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ['3.9', '3.10']
    timeout-minutes: 20
    defaults:
      run:
        working-directory: unitycatalog-ai
    steps:
      - name: Checkout code
        uses: actions/checkout@v4
      - name: Set up Python
        uses: actions/setup-python@v5
        with:
          python-version: ${{ matrix.python-version }}
      - name: Install dependencies
        run: |
          pip install .
          pip install integrations/anthropic[dev]
      - name: Run tests
        run: |
          pytest integrations/anthropic/tests

  openai_test:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ['3.9', '3.10']
    timeout-minutes: 20
    defaults:
      run:
        working-directory: unitycatalog-ai
    steps:
      - name: Checkout code
        uses: actions/checkout@v4
      - name: Set up Python
        uses: actions/setup-python@v5
        with:
          python-version: ${{ matrix.python-version }}
      - name: Install dependencies
        run: |
          pip install .
          pip install integrations/openai[dev]
      - name: Run tests
        run: |
          pytest integrations/openai/tests
```

**Comment by BenWilson2** [comment]
@dennyglee ooooh good catch. Our unit test suite should have 

```yaml
permissions:
  contents: read
```

Before the jobs: definition. I don't have the ability to edit this PR, though. LMK if you would prefer for me to re-file and fix while Serena is out

**Review by dennyglee — APPROVED** [review]

**Comment by dennyglee** [comment]
@BenWilson2 could you validate this real quick before I merge as I had just resolved a conflict.

File: .github/workflows/ucai-integration-tests.yml
Comment (BenWilson2) [reply .github/workflows/ucai-integration-tests.yml:85]: Sorry, my mistake on the suggestion code block

**Review by BenWilson2 — APPROVED** [review]
Looks good!

**Pull request merged**
